File: js/examples/basic.js
const { AIGodfather } = require('aigodfather')

const ai = new AIGodfather({
  apiKey: process.env.AIGODFATHER_API_KEY,
  debug: true
})

async function main() {
  // Check connection
  const status = await ai.ping()
  console.log(`Connected: ${status.agentName}`)
  console.log(`Events remaining: ${status.limits.remaining}`)

  // Track events
  await ai.info('App started', { version: '1.0.0' })
  await ai.warning('High memory', { percent: 87 })
  await ai.error('Payment failed', { orderId: 'ord_123' })
  await ai.critical('Database down')

  // Custom event
  await ai.track('user_signup', {
    severity: 'low',
    message: 'New user registered',
    metadata: { plan: 'trial', country: 'RO' },
    userId: 'user_456',
    tags: ['signup']
  })
}

main().catch(console.error)

File: python/examples/basic.py
import os
from aigodfather import AIGodfather

ai = AIGodfather(
    api_key=os.environ['AIGODFATHER_API_KEY'],
    debug=True
)

# Check connection
status = ai.ping()
print(f"Connected: {status['agentName']}")
print(f"Events remaining: {status['limits']['remaining']}")

# Track events
ai.info('App started', {'version': '1.0.0'})
ai.warning('High memory', {'percent': 87})
ai.error('Payment failed', {'order_id': 'ord_123'})
ai.critical('Database down')

# Custom event
ai.track(
    'user_signup',
    severity='low',
    message='New user registered',
    metadata={'plan': 'trial', 'country': 'RO'},
    user_id='user_456',
    tags=['signup']
)
